File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/dtos/MedicineDto.java
package com.example.pharmadicmobile.dtos;

import com.google.gson.annotations.SerializedName;

public class MedicineDto {
    @SerializedName(value = "MedicineId", alternate = {"medicineId"})
    private int medicineId;

    @SerializedName(value = "MedicineName", alternate = {"medicineName"})
    private String medicineName;

    @SerializedName(value = "CategoryName", alternate = {"categoryName"})
    private String categoryName;

    @SerializedName(value = "DosageForm", alternate = {"dosageForm"})
    private String dosageForm;

    @SerializedName(value = "Manufacturer", alternate = {"manufacturer"})
    private String manufacturer;

    @SerializedName(value = "Uses", alternate = {"uses"})
    private String uses;

    @SerializedName(value = "Dosage", alternate = {"dosage"})
    private String dosage;

    @SerializedName(value = "Contraindications", alternate = {"contraindications"})
    private String contraindications;

    @SerializedName(value = "ImageUrl", alternate = {"imageUrl"})
    private String imageUrl;

    public int getMedicineId() { return medicineId; }
    public String getMedicineName() { return medicineName; }
    public String getCategoryName() { return categoryName; }
    public String getDosageForm() { return dosageForm; }
    public String getManufacturer() { return manufacturer; }
    public String getUses() { return uses; }
    public String getDosage() { return dosage; }
    public String getContraindications() { return contraindications; }
    public String getImageUrl() { return imageUrl; }
}

File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/DTOs/MedicineDto.cs
namespace PharmaDicBackEnd.ApiService.DTOs
{
    public class MedicineDto
    {
        public int MedicineId { get; set; }
        public string MedicineName { get; set; } = null!;
        public string? CategoryName { get; set; }
        public string? DosageForm { get; set; }
        public string? Manufacturer { get; set; }
        public string? Uses { get; set; }

        public string Sku => $"PH-TH-{MedicineId:D3}";
        public DateTime? UpdatedAt { get; set; }
        public string? UpdatedBy { get; set; }
    }
}